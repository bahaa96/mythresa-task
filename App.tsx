import './src/locale/config';
import './src/utils';

import * as eva from '@eva-design/eva';
import {
  ApplicationProvider,
  IconRegistry,
  Layout,
} from '@ui-kitten/components';
import { EvaIconsPack } from '@ui-kitten/eva-icons';
import * as SplashScreen from 'expo-splash-screen';
import { StatusBar } from 'expo-status-bar';
import React, { useEffect } from 'react';
import { StyleSheet } from 'react-native';
import { QueryClient, QueryClientProvider } from 'react-query';

import { FavoritesContextProvider } from './src/context/FavoritesContext/FavoritesContext';
import { LocaleContextProvider } from './src/context/LocaleContext';
import Router from './src/router';
import theme from './src/theme';

const queryClient = new QueryClient();

const App: React.FC = () => {
  useEffect(() => {
    (async () => {
      // Prevent native splash screen from auto hiding
      // This shows up a warning due to an expo deprecation issue.
      try {
        await SplashScreen.preventAutoHideAsync();
      } catch (e) {
        console.warn(e);
      }
    })();
  }, []);

  return (
    <QueryClientProvider client={queryClient}>
      <StatusBar style="dark" />
      <IconRegistry icons={EvaIconsPack} />
      <ApplicationProvider {...eva} theme={{ ...eva.light, ...theme }}>
        <Layout style={styles.container}>
          <LocaleContextProvider>
            <FavoritesContextProvider>
              <Router />
            </FavoritesContextProvider>
          </LocaleContextProvider>
        </Layout>
      </ApplicationProvider>
    </QueryClientProvider>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
export default App;
