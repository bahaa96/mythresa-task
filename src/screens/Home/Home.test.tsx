import { renderHook } from '@testing-library/react-hooks';
import { render } from '@testing-library/react-native';
import { request } from 'http';
import nock from 'nock';
import React from 'react';
import { QueryClient, QueryClientProvider, useQuery } from 'react-query';

import App from '../../../App.tsx';

describe('it should render the app properly', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<App />);
    expect(toJSON()).toMatchSnapshot();
  });
});

const useFetchData = () => {
  return useQuery('movies', () => {
    return request('/api/data');
  });
};

describe('should render the movie list', () => {
  it('should render the fetched data', async () => {
    const queryClient = new QueryClient();
    const wrapper = ({ children }) => {
      return (
        <QueryClientProvider client={queryClient}>
          {children}
        </QueryClientProvider>
      );
    };

    const expectation = nock('http://example.com').get('/api/data').reply(200, {
      answer: 42,
    });

    const { result, waitFor } = renderHook(
      () => {
        return useFetchData();
      },
      { wrapper },
    );

    await waitFor(() => {
      console.log('result: ', result.current);
      return result.current.isSuccess;
    });
    expect(result.current).toEqual({ answer: 42 });
  });
});
