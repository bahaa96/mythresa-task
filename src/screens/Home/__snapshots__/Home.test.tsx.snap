// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render the app properly renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    appearance="default"
    level="1"
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
        },
        Object {
          "flex": 1,
        },
      ]
    }
  />
</View>
`;

exports[`it works renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    appearance="default"
    level="1"
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
        },
        Object {
          "flex": 1,
        },
      ]
    }
  />
</View>
`;
